Automattic/wp-calypso
# Jetpack Manage: Add user feedback modal form.

This pull request adds a modal form for user feedback. It can be accessed by clicking the **'Share user feedback'** option in the sidebar.

<img width="851" alt="Screen Shot 2024-01-03 at 3 09 28 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/de2c8633-70a7-4c02-8fb1-97136c8fd3d2">

**Note: The submit button is not functional and will be addressed in a separate PR.**

Closes https://github.com/Automattic/jetpack-genesis/issues/163

## Proposed Changes

* Implement the User feedback form modal based on the `Modal` component.
* Add 'Click' callback in the Share feedback button to show the modal.

## Testing Instructions
Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb.

* Use the Jetpack cloud live link below and go to the Dashboard page (`/dashboard?flags=jetpack/user-feedback-form`)
* Click the **'Share user feedback'** menu item in the sidebar.
<img width="292" alt="screen-shot-2023-12-20-at-7 39 21-pm" src="https://github.com/Automattic/wp-calypso/assets/56598660/eef6bf7f-7106-43aa-ad02-0d83b3340313">

* Confirm the modal shows up. Verify that the components within the modal work as expected.

## Pre-merge Checklist
